src/Formatter/HumanReadableExceptionFormatter.php

```diff
@@ -0,0 +1,161 @@
+    {
+        $lines = [];
+        $lines[] = $this->printFirstLineFromException($exception, $nested);
+
+        foreach ($exception->getTrace() as $traceLine) {
+            $lines[] = $this->formatTraceLine($traceLine);
+        }
+
+        if ($exception->getPrevious()) {
+            $previousLines = $this->linesFromException($exception->getPrevious(), true);
+
+            // recursion makes sure deeper nesting has longer prefixes
+            foreach ($previousLines as $previousLine) {
+                $lines[] = '……' . $previousLine;
+            }
+        }
+
+        return $lines;
+    }
+
+    protected function printFirstLineFromException(\Throwable $exception, bool $nested): string
+    {
+        return sprintf('[%s] %s: %s in %s:%s',
+            // only the root exception has a timestamp
+            $nested ? 'Previous Exception' : date('r'),
+            get_class($exception),
+            $exception->getMessage(),
+            $exception->getFile(),
+            $exception->getLine()
+        );
+    }
+
+    /**
+     * Transform an element of the array Exception::getTrace
+     * to a string of a single line
+     *
+     * @param mixed[] $trace element of Exception::getTrace
+     * @return string of a single line
+     */
+    protected function formatTraceLine(array $trace): string
+    {
+        $arguments = $this->formatArguments($trace['args']);
+        $output = '';
+
+        // only display file if we're not in the context of a class
+        // to save space for the sake for readability of the stack trace
+        if (isset($trace['file']) && !isset($trace['class'])) {
+            $output .= $trace['file'];
+
+            if (isset($trace['line'])) {
+                $output .= '(' . $trace['line'] . ')';
+            }
+
+            $output .= ': ';
+        }
+
+        $output .= $trace['class'] ?? '';
+        $output .= $trace['type'] ?? '';
+        $output .= $trace['function'] ?? '';
+
+        $arguments = $this->formatArguments($trace['args'] ?? []);
+        $output .= '(' . $arguments . ')';
+
+        if (isset($trace['class']) && isset($trace['line'])) {
+            $output .= ':' . $trace['line'];
+        }
+
+        return $output;
+    }
+
+    /**
+     * Convert function arguments of any type to a short readable string
+     *
+     * @param mixed[] $arguments
+     * @return string a summary of the list of arguments
+     */
+    protected function formatArguments(array $arguments): string
+    {
+        return implode(', ', array_map([$this, 'formatArgument'], $arguments));
+    }
+
+    /**
+     * Summarize a variable
+     *
+     * @param mixed $argument anything at all
+     * @return string a summary of the argument
+     */
+    protected function formatArgument($argument): string
+    {
+        if (is_int($argument) || is_float($argument)) {
+            return $argument;
+        }
+
+        if (is_string($argument)) {
+            if (strlen($argument) < 80) {
```

Comment:
Why would you like to truncate this?

Reply:
I don't want e.g. large binaries or large XML strings in my logs.